src/main/java/net/sf/jabref/gui/openoffice/OOBibBase.java

```diff
@@ -1296,4 +1297,31 @@ public int hashCode() {
 
     }
 
+
+    public BibDatabase generateDatabase(List<BibDatabase> databases, OOBibStyle style)
+            throws NoSuchElementException, WrappedTargetException {
+        BibDatabase resultDatabase = new BibDatabase();
+        List<String> cited = findCitedKeys();
+        for (String key : cited) {
+            for (BibDatabase loopDatabase : databases) {
+                Optional<BibEntry> entry = loopDatabase.getEntryByKey(key);
+                if (entry.isPresent()) {
+                    BibEntry clonedEntry = (BibEntry) entry.get().clone();
+                    clonedEntry.setId(IdGenerator.next());
+                    resultDatabase.insertEntry(clonedEntry);
+                    entry.get().getFieldOptional(FieldName.CROSSREF).ifPresent(crossref -> {
+                        if (!resultDatabase.getEntryByKey(crossref).isPresent()) {
+                            Optional<BibEntry> refEntry = loopDatabase.getEntryByKey(crossref);
+                            if (refEntry.isPresent()) {
+                                resultDatabase.insertEntry(refEntry.get());
```

Comment:
Makes sense. Copied the code from the AUX parser where more things were done in the corresponding place and didn't reflect on it.